webapp/channels/src/packages/mattermost-redux/src/reducers/entities/scheduled_posts.ts

```diff
@@ -22,6 +22,20 @@ function byTeamId(state: ScheduledPostsState['byTeamId'] = {}, action: AnyAction
 
         return newState;
     }
+    case ScheduledPostTypes.SINGLE_SCHEDULED_POST_RECEIVED: {
+        const scheduledPost = action.data.scheduledPost;
+        const teamId = action.data.teamId || 'directChannels';
+
+        const newState = {...state};
+
+        if (newState[teamId]) {
+            newState[teamId] = [...newState[teamId], scheduledPost];
+        } else {
+            newState[teamId] = [scheduledPost];
+        }
+
+        return newState;
+    }
```

Comment:
Probably not, but do we want to check for repetitions?

One thing that could happen (depending on how we implement it) is that we create the scheduled post locally, but the server send us the websocket event either way. In that case, we would add the post twice.

This could be handled somewhere else and not in the reducer, so 0/5.

Reply:
I'm sending the connection ID to the server, and server won't send the websocket event to this connection. This actually solves this precise problem. But anyway, I added the check for duplicate scheduled post by ID and removed it as it may help at some point.